sktime/forecasting/darts.py

```diff
@@ -0,0 +1,232 @@
+        multi_models: Optional[bool] = True,
+        use_static_covariates: Optional[bool] = True,
+        kwargs: Optional[dict] = None,
+    ) -> None:
+        self.lags = lags
+        self.lags_past_covariates = lags_past_covariates
+        self.lags_future_covariates = lags_future_covariates
+        self.output_chunk_length = output_chunk_length
+        self.add_encoders = add_encoders
+        self.likelihood = likelihood
+        self.quantiles = quantiles
+        self.random_state = random_state
+        self.multi_models = multi_models
+        self.use_static_covariates = use_static_covariates
+        self.kwargs = kwargs
+        self.handle_kwargs()
+
+        super().__init__(past_covariates=past_covariates, num_samples=num_samples)
+
+    def handle_kwargs(self: "DartsXGBModel") -> None:
+        """Handle additional keyword arguments."""
+        if self.kwargs is not None:
+            for key, value in self.kwargs.items():
+                setattr(self, key, value)
+
+    def _create_forecaster(self: "DartsXGBModel"):
+        """Create Darts model."""
+        from darts.models import XGBModel
+
+        return XGBModel(
+            lags=self.lags,
+            lags_past_covariates=self.lags_past_covariates,
+            lags_future_covariates=self.lags_future_covariates,
+            output_chunk_length=self.output_chunk_length,
+            add_encoders=self.add_encoders,
+            likelihood=self.likelihood,
+            quantiles=self.quantiles,
+            random_state=self.random_state,
+            multi_models=self.multi_models,
+            use_static_covariates=self.use_static_covariates,
+            kwargs=self.kwargs,
+        )
+
+    @property
+    def _lags_past_covariates(
+        self: "DartsXGBModel",
+    ) -> Union[int, List[int], Dict[str, Union[int, List[int]]], None]:
+        return self.lags_past_covariates
+
+    @property
+    def _lags_future_covariates(
+        self: "DartsXGBModel",
+    ) -> Union[
+        Tuple[int, int], List[int], Dict[str, Union[Tuple[int, int], List[int]]], None
+    ]:
+        return self.lags_future_covariates
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+        """
+        del parameter_set  # to avoid being detected as unused by ``vulture`` etc.
+
+        if _check_soft_dependencies(
+            "u8darts", package_import_alias={"u8darts": "darts"}, severity="error"
```

Comment:
You need to use `none` instead of `error`. Currently if it's missing, it's going to raise an error directly and that's causing the issue in CI. `none` is going to return a boolean.

Reply:
yea sure let me change that